web/packages/teleport/src/Discover/Server/DiscoveryConfigSsm/DiscoveryConfigSsm.tsx

```diff
@@ -0,0 +1,372 @@
+              <Link
+                target="_blank"
+                href="https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AmazonSSMManagedInstanceCore.html"
+              >
+                AmazonSSMManagedInstanceCore
+              </Link>{' '}
+              policy to EC2 instances IAM profile. The policy enables EC2
+              instances to use SSM core functionality.
+            </Text>
+          </StyledBox>
+          <StyledBox mt={4}>
+            <Text bold>Step 3</Text>
+            Each EC2 instance requires{' '}
+            <Link
+              target="_blank"
+              href="https://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent-status-and-restart.html"
+            >
+              SSM Agent
+            </Link>{' '}
+            to be running. The SSM{' '}
+            <Link
+              target="_blank"
+              href={`https://${selectedRegion}.console.aws.amazon.com/systems-manager/fleet-manager/managed-nodes?region=${selectedRegion}`}
+            >
+              Nodes Manager dashboard
+            </Link>{' '}
+            will list all instances that have SSM agent already running. Ensure
+            ping statuses are <Mark>Online</Mark>.
+            <Info mt={3} mb={0}>
+              If you do not see your instances listed in the dashboard, it might
+              take up to 30 minutes for your instances to use the IAM
+              credentials you updated in step 2.
+            </Info>
+          </StyledBox>
+          <Validation>
+            {({ validator }) => (
+              <StyledBox mt={4}>
+                <Text bold>Step 4</Text>
+                <Box>
+                  <Text typography="subtitle1" mb={1}>
+                    Give a name for the{' '}
+                    <Link
+                      target="_blank"
+                      href="https://docs.aws.amazon.com/systems-manager/latest/userguide/documents.html"
+                    >
+                      AWS SSM Document
+                    </Link>{' '}
+                    that will be created on your behalf. Required to run the
+                    installer script on each discovered instances.
+                  </Text>
+                  <FieldInput
+                    rule={requiredSsmDocument}
+                    label="SSM Document Name"
+                    value={ssmDocumentName}
+                    onChange={e => setSsmDocumentName(e.target.value)}
+                    placeholder="ssm-document-name"
+                    disabled={!!scriptUrl}
+                  />
+                </Box>
+                <ButtonSecondary
+                  onClick={() =>
+                    scriptUrl ? setScriptUrl('') : generateScriptUrl(validator)
+                  }
+                  disabled={!selectedRegion}
+                >
+                  {scriptUrl ? 'Edit' : 'Next'}
+                </ButtonSecondary>
+              </StyledBox>
+            )}
+          </Validation>
```

Comment:
This could be an actual `<form>` with `onSubmit` and with `type="submit"` on the button. This way it could be submitted either by clicking the button or inputting some text in the field and then pressing enter.